### DISCUSSION TITLE
Error running lammps

### DISCUSSION THREAD

**Discussion by Xi-yuanWang** [deepmodeling/deepmd-kit#249] [Q&A]
I successfully compile the lammps stable_7Aug2019 and its deepmd-kit module , but when I try to use dpgen, lammps throw the following error.

LAMMPS (3 Mar 2020)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:94)
  using 1 OpenMP thread(s) per MPI task
Reading data file ...
  triclinic box = (0 0 0) to (9.95439 9.80314 9.83274) with tilt (0.248331 0.192711 -0.147738)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  5 atoms
  read_data CPU = 0.000724077 secs
Changing box ...
  triclinic box = (0 0 0) to (9.95439 9.80314 9.83274) with tilt (0.248331 0.192711 -0.147738)
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /nfs-share/home/1800011848/miniconda3
  source:             v1.1.4-210-gbc01883
  source brach:       HEAD
  source commit:      bc01883
  source commit at:   2020-05-23 23:34:57 +0800
  build float prec:   double
  build with tf inc:  /nfs-share/home/1800011848/miniconda3/include;/nfs-share/home/1800011848/miniconda3/include
  build with tf lib:  /nfs-share/home/1800011848/miniconda3/lib/libtensorflow_cc.so;/nfs-share/home/1800011848/miniconda3/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /nfs-share/home/1800011848/miniconda3
  source:             v1.1.4-210-gbc01883
  source branch:      HEAD
  source commit:      bc01883
  source commit at:   2020-05-23 23:34:57 +0800
  build float prec:   double
  build with tf inc:  /nfs-share/home/1800011848/miniconda3/include;/nfs-share/home/1800011848/miniconda3/include
  build with tf lib:  /nfs-share/home/1800011848/miniconda3/lib/libtensorflow_cc.so;/nfs-share/home/1800011848/miniconda3/lib/libtensorflow_framework.so
2020-09-03 07:59:11.057731: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
2020-09-03 07:59:11.079920: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2600000000 Hz
2020-09-03 07:59:11.082976: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55c7aff28cd0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-09-03 07:59:11.083037: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
**Invalid argument: Input 0 of node DescrptSeA was passed float from Reshape_2:0 incompatible with expected double.**

The input file is
variable        NSTEPS          equal 300
variable        THERMO_FREQ     equal 10
variable        DUMP_FREQ       equal 10
variable        TEMP            equal 100.000000
variable        PRES            equal -1.000000
variable        TAU_T           equal 0.100000
variable        TAU_P           equal 0.500000

units           metal
boundary        p p p
atom_style      atomic

neighbor        1.0 bin

box          tilt large
read_data       conf.lmp
change_box   all triclinic
mass            1 1.000000
mass            2 12.000000
pair_style      deepmd ../graph.001.pb ../graph.002.pb ../graph.003.pb ../graph.000.pb  out_freq ${THERMO_FREQ} out_file model_devi.out
pair_coeff

thermo_style    custom step temp pe ke etotal press vol lx ly lz xy xz yz
thermo          ${THERMO_FREQ}
dump            1 all custom ${DUMP_FREQ} traj/*.lammpstrj id type x y z

velocity        all create ${TEMP} 681862
fix             1 all nvt temp ${TEMP} ${TEMP} ${TAU_T}

timestep        0.002000
run             ${NSTEPS}

**Comment by njzjz** [comment]
You are using the double precision LAMMPS with a float precision model. If the model is not what you want, please check the installation of deepmd-kit.

**Comment by Xi-yuanWang** [comment]
Then how to control the precision of a model or which model is in double precision?

**Comment by njzjz** [comment]
I guess you installed with `conda install`? If you executed `conda install deepmd-kit=*=*cpu -c deepmodeling`, it's double precision; if you executed `conda install deepmd-kit=*=*cpu_low -c deepmodeling`, that's float precision.

If you used `pip` to install and didn't change `setup.py`, it should be double precision:
https://github.com/deepmodeling/deepmd-kit/blob/567bcff2a99ab50b9b1ba5066575da1a99a4c1b8/setup.py#L58

**Reply by konokcbhowmik** [reply]
Sir, I am facing the problem in windows 10.
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task

What should I do?

**Comment by Xi-yuanWang** [comment]
It works，thank you.
